### PR TITLE
Run2-sim13 Backport PR #19478 for fixing bug of ECAL simulation

### PR DESCRIPTION
**Pull request by bsunanda**
Fix the bug in the calculation of the distance of the hit point to the readout unit

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
@bsunanda, I think, that a flag should b added to this PR allowing to switch between default and fixed variants.

**Comment by bsunanda** [comment]
@civanch I have added the flag g4SimHits.ECalSD.IgnoreDepthCorr which is True as default and can be set to False to make the correct depth estimation

**Comment by civanch** [comment]
+1

File: SimG4CMS/Calo/src/ECalSD.cc [hunk 1]
```diff
@@ -225,14 +226,13 @@ int ECalSD::getTrackID(G4Track* aTrack) {
...
   G4LogicalVolume* lv   = aStep->GetPreStepPoint()->GetTouchable()->GetVolume(0)->GetLogicalVolume();
-  uint16_t ret = 0;
-  if (any(useDepth1,lv))      ret = 1;
-  else if (any(useDepth2,lv)) ret = 2;
-  else if (storeRL) ret = getRadiationLength(aStep);
-#ifdef DebugLog
-  LogDebug("EcalSim") << "Volume " << lv->GetName() << " Depth " << ret;
-#endif
-  return ret;
+  if      (storeRL) {
```
Comment (davidlange6) [inline SimG4CMS/Calo/src/ECalSD.cc]: hi @bsunanda @civanch - the reordering of this if statement is naively a behavior change. What is the purpose?

**Comment by perrozzi** [comment]
@bsunanda this PR is preventing the release of a new CMSSW 71X containing important developments, already merged, needed for urgent GEN productions.
May I kindly ask to finalize this PR asap?
Thanks a lot in advance

**Comment by bsunanda** [comment]
@perrozzi @civanch As far as I can see this PR is up-to-date and has no issue. It is waiting for Vladimir's confirmation. In the last ORP Vladimir mentioned that he will do that in a day or two.

**Comment by civanch** [comment]
@davidlange6 , I asked at the meeting if a simple python fragment is needs to be added in order be able more easy enable new feature. I cannot add such fragment before this PR is merged.

**Comment by davidlange6** [comment]
you can actually add it ( of course it won't work..but at least things can get done in parallel)

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch @davidlange6 I have added the changes so that one can make simulation keeping radiation length information and then doing digitization & reconstruction later on.

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**